### DISCUSSION TITLE
Possibility to start programs which aren't available through the public API

### DISCUSSION THREAD

**Discussion by mr-p666** [ekutner/home-connect-hass#285] [Ideas]
Hello folks,
I wonder whether it's possible to start programs via this integration, even if they are only available with the proprietary API.
In the case of my Siemens dishwasher, this could work because I always preselected the program (Intelligent), which isn't available via the public API.  It's currently not possible to start the dishwasher without first selecting a program. But if the API allows you to simply send the start command with the program "Last" (i.e. effectively without a selected program), it might be possible to accomplish the process this way, as long as the API's start command doesn't have a selected program as parameter expected.

**Comment by mr-p666** [comment]
Is nobody interested in this function or can at least anyone judge whether it would be possible in principle?

**Comment by matthewcbyington** [comment]
I had a similar question. I defined a custom cycle on my Bosch dishwasher that has all the settings I like. I'd like to be able to Start that cycle from the HA app, but it's not available in the list of selected programs on the input_select. It does show as the selected program though (
<img width="651" alt="image" src="https://github.com/ekutner/home-connect-hass/assets/1623447/ac5687c3-2f31-4f1e-9c3f-a2349e1a2cad">

**Comment by ekutner** [comment]
I can't test it to verify but it might work with the start_program service. Make sure not to specify a program and set validate to false.

**Reply by mr-p666** [reply]
Sounds good in principle.
Maybe this is a stupid question, but how do you disable verification? :smiley:

**Reply by ekutner** [reply]
the service has a "validate" parameter, set it to false.

**Reply by mr-p666** [reply]
Unfortunately it doesn't allow the service execution without a valid program_key.
![image](https://github.com/ekutner/home-connect-hass/assets/4217530/0f8d1665-dee6-4901-8fe2-7c97cc693b4e)

**Reply by ekutner** [reply]
not very surprising

**Comment by JWdeBokx** [comment]
My Neff oven has something like this.
It has a program displayed in the state as Cooking.Oven.Program.HeatingMode.BreadBaking
But can not select it as a program.
I can select in the homeconnect app.

I will try the start_program service later and let it know....

**Reply by JWdeBokx** [reply]
Tested the service, and indeed it's not accepting the value.
On the site is stated:
The following list contains all oven programs, which are supported by the API. Please note that not all programs are supported for a specific oven model. Therefore, we recommend to use the [available program interface](https://apiclient.home-connect.com/#/programs/get_available_programs) to check the program availability.

So not possible I think.
Same problem with the dishwasher I suppose.

But how is it possible the homeconnect app does support it?
Are they using an api not public available?

**Reply by ekutner** [reply]
The HC mobile app is using a private API and can even connect directly to the appliances when located on the same network, without going through the HC cloud API. This isn't documented and not supported for 3rd party apps.

**Reply by JWdeBokx** [reply]
Hmm too bad...
But a problem with many appliances...
Didn't know about the direct connecting.
Without cloud is too be preferred...
